symbol-table/src/types/functions/function_input.rs

```diff
@@ -52,10 +58,34 @@ impl FunctionInputType {
     pub fn span(&self) -> &Span {
         match self {
             FunctionInputType::InputKeyword(identifier) => &identifier.span,
+            FunctionInputType::SelfKeyword(identifier) => &identifier.span,
+            FunctionInputType::MutSelfKeyword(identifier) => &identifier.span,
             FunctionInputType::Variable(variable) => &variable.span,
         }
     }
 
+    ///
+    /// Returns `true` if input `self` or `mut self` is present.
+    /// Returns `false` otherwise.
+    ///
+    pub fn is_self(&self) -> bool {
+        match self {
+            FunctionInputType::InputKeyword(_) => false,
+            FunctionInputType::SelfKeyword(_) => true,
+            FunctionInputType::MutSelfKeyword(_) => true,
+            FunctionInputType::Variable(_) => false,
+        }
+    }
+
+    ///
+    /// Returns `0` if the function input is a `self` or `mut self` keyword which does not have to
+    /// provided in a call to the function.
+    /// Returns `1` if a variable must be provided in a call to the function.
+    ///
+    pub fn count(&self) -> usize {
```

Comment:
this method is a bit misleading, as there's already an `Iterator::count` in the `std`; I'll recommend an alternative at the only callsite where I see it used